app/src/main/java/com/example/androidtrainingproject/login/LoginActivity.kt

```diff
@@ -88,20 +87,26 @@ fun LoginScreen(navigateToProductDetails: () -> Unit = {}) {
             },
         )
 
-        if (showWrongPasswordMessage.value) {
-            Text("Invalid login credentials", color = Color.Red)
+        loginViewModel.isLoginSuccessful?.let { isLoggedIn ->
+            if (isLoggedIn) {
+                navigateToProductDetails()
+                Toast.makeText(LocalContext.current, "Login successful", Toast.LENGTH_SHORT).show()
+                loginViewModel.isLoginSuccessful = null
+            } else {
+                Toast.makeText(LocalContext.current, "Invalid login credentials", Toast.LENGTH_SHORT).show()
+                loginViewModel.isLoginSuccessful = null
+            }
         }
 
         Button(
             modifier = Modifier
                 .padding(horizontal = 20.dp)
                 .fillMaxWidth(),
+            enabled = !loginViewModel.isLoading,
             onClick = {
-                if (usernameField.value == hardcodedUsername && passwordField.value == hardcodedPassword) {
-                    navigateToProductDetails()
-                } else {
-                    showWrongPasswordMessage.value = true;
-                }
+                loginViewModel.login(usernameField.value, passwordField.value)
+                loginViewModel.isLoading = true
```

Comment:
This should be done as part of the login method, no need to request it separately, you might delete it accidentally.

Reply:
Fair point, moved it to the login's try/catch